src/measure_theory/interval_integral.lean

```diff
@@ -0,0 +1,405 @@
+
+lemma integral_self : (∫ x in a...a, f x) = 0 :=
+by { rw if_pos (le_refl _), apply integral_on_volume_zero, rw [interval_self, volume_singleton] }
+
+lemma integral_of_le (h : a ≤ b) (f : ℝ → β) :
+  (∫ x in a...b, f x) = ∫ x in a..b, f x :=
+if_pos h
+
+lemma integral_of_not_le (h : ¬ a ≤ b) (f : ℝ → β) :
+  (∫ x in a...b, f x) = - ∫ x in a..b, f x :=
+if_neg h
+
+lemma integral_of_le' (h : b ≤ a) (f : ℝ → β) :
+  (∫ x in a...b, f x) = - ∫ x in a..b, f x :=
+begin
+  split_ifs with h',
+  { rw [le_antisymm h h', integral_on_interval_self, neg_zero] },
+  refl
+end
+
+lemma integral_eq_neg_swap (a b : ℝ) : (∫ x in a...b, f x) = - (∫ x in b...a, f x) :=
+begin
+  cases le_total a b with ab ba,
+  { rw [integral_of_le ab, integral_of_le' ab, interval_swap, neg_neg] },
+  { rw [integral_of_le ba, integral_of_le' ba, interval_swap] }
+end
+
+lemma integral_undef (h : ¬ (measurable_on a..b f ∧ integrable_on a..b f)) :
+  (∫ x in a...b, f x) = 0 :=
+begin
+  split_ifs with hab,
+  { simp only [interval_of_le hab] at *, exact integral_undef h },
+  { simp only [interval_of_not_le hab] at *, rw [integral_undef h, neg_zero] }
+end
+
+lemma integral_congr (h : ∀ x, x ∈ a..b → f x = g x) : (∫ x in a...b, f x) = ∫ x in a...b, g x :=
+begin
+  split_ifs with h',
+  { simp only [interval_of_le h'] at *,
+    exact integral_on_congr h },
+  { simp only [interval_of_not_le h'] at *,
+    rw integral_on_congr h }
+end
+
+lemma integral_congr_ae (h : ∀ₘ x, x ∈ a..b → f x = g x)
+  (hfm : measurable_on a..b f) (hgm : measurable_on a..b g) :
+  (∫ x in a...b, f x) = (∫ x in a...b, g x) :=
+begin
+  split_ifs with h,
+  { simp only [interval_of_le h] at *,
+    rw integral_on_congr_of_ae_eq,
+    assumption' },
+  { simp only [interval_of_not_le h] at *,
+    rw integral_on_congr_of_ae_eq,
+    assumption' }
+end
+
+lemma integral_add (hfm : measurable_on a..b f) (hfi : integrable_on a..b f)
+  (hgm : measurable_on a..b g) (hgi : integrable_on a..b g) :
+  (∫ x in a...b, f x + g x) = (∫ x in a...b, f x) + (∫ x in a...b, g x) :=
+begin
+  split_ifs with h,
+  { simp only [interval_of_le h] at *,
+    rw [integral_on_add], assumption' },
+  { simp only [interval_of_not_le h] at *,
+    rw [← neg_add, integral_on_add], assumption' }
+end
+
+lemma integral_const {c : β} : (∫ x in a...b, c) = (b - a) • c :=
+begin
+  split_ifs with h,
+  { rw [integral_on_const, volume_interval, ← ennreal.of_real, ennreal.to_real_of_real, abs_of_nonneg],
+    exact sub_nonneg_of_le h, exact abs_nonneg _, exact is_measurable_interval },
+  { rw [integral_on_const, volume_interval, ← ennreal.of_real, ennreal.to_real_of_real,
+      abs_of_nonpos, ← neg_smul, neg_neg],
+    exact sub_nonpos_of_le (le_of_not_ge h), exact abs_nonneg _, exact is_measurable_interval }
+end
+
+lemma norm_integral_le_abs (a b : ℝ) (f : ℝ → β):
+  ∥(∫ x in a...b, f x)∥ ≤ abs (∫ x in a...b, ∥f x∥) :=
```

Comment:
Do you prove `∥(∫ x in a...b, f x)∥ ≤ ∫ x in a..b, ∥f x∥` as well (and equality of these RHSs)?

Reply:
It will be easier to prove trivial parts of some of these theorems if you have `abs (∫ x in a...b, f x) = ∫ x in a..b, f x` whenever `f x ≥ 0` on `a..b`.